app/controllers/check_ins_controller.rb

```diff
@@ -15,6 +15,19 @@ def create
     end
   end
 
+  def edit
+    @check_in = current_account.check_ins.find(params[:id])
+  end
+
+  def update
+    @check_in = current_account.check_ins.find(params[:id])
+    log_ids = check_in_params[:log_entries_attributes].map { |_, attrs| attrs[:id] }.compact
+    raise "YOU DID A BAD!" if current_account.logs.where(id: log_ids).count != log_ids.count
```

Comment:
Let's extract these two lines to their own method: `authorize_current_account_may_update_log_entries`